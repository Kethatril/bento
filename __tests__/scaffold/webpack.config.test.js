import { join, resolve } from "path";
import shelljs from "shelljs";
import tmp from "tmp";

describe("webpack.config.js", () => {
    const cwd = join(resolve(__dirname, "../../"));

    afterEach(() => {
        process.chdir(cwd);
    });

    it("options to build webpack.config.js matches snapshot for Single Page Application", () => {
        const { name: dir } = tmp.dirSync();

        shelljs.cd(dir);
        shelljs.exec(`ln -s ${cwd}/node_modules`);
        shelljs.exec("mkdir -p ./node_modules/@gladeye");
        shelljs.cd("./node_modules/@gladeye");
        shelljs.exec(`ln -s ${cwd}`);
        shelljs.cd(dir);
        shelljs.exec(
            "npx yo ./node_modules/@gladeye/bento/scaffold --default spa --no-confirm --quiet"
        );
        const options = require(join(dir, "./app/config/webpack"));
        options.paths.root = "<tmp>";
        expect(options).toMatchSnapshot();
    });

    it("options to build webpack.config.js matches snapshot for Server Side Application", () => {
        const { name: dir } = tmp.dirSync();

        shelljs.cd(dir);
        shelljs.exec(`ln -s ${cwd}/node_modules`);
        shelljs.exec("mkdir -p ./node_modules/@gladeye");
        shelljs.cd("./node_modules/@gladeye");
        shelljs.exec(`ln -s ${cwd}`);
        shelljs.cd(dir);
        shelljs.exec(
            "npx yo ./node_modules/@gladeye/bento/scaffold --default ssa --no-confirm --quiet"
        );
        const options = require(join(dir, "./assets/config/webpack"));
        options.paths.root = "<tmp>";
        expect(options).toMatchSnapshot();
    });
});
