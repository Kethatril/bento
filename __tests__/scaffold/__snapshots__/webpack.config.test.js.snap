// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack.config.js options to build webpack.config.js matches snapshot for Server Side Application 1`] = `
Object {
  "babel": Object {
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": "@{autoprefixer.browsers}",
          },
          "useBuiltIns": true,
        },
      ],
      "stage-2",
    ],
  },
  "blocks": Object {
    "list": Array [
      [Function ports],
      [Function filename],
      [Function base],
      [Function script],
      [Function style],
      [Function media],
      [Function plugins],
      [Function devServer],
      [Function backEnd],
    ],
    "timeout": 3000,
  },
  "browserSync": Object {
    "files": Array [],
    "ghostMode": false,
    "open": true,
    "watchOptions": Object {
      "cwd": "@{paths.root}",
      "ignoreInitial": true,
      "ignored": "*.txt",
    },
  },
  "caching": Object {
    "hash": ".[hash:8]",
  },
  "enabled": Object {
    "sourceMap": "@{env.isDevServer}",
  },
  "entry": Object {
    "main": Array [
      "./scripts/main.js",
    ],
  },
  "env": Object {
    "isDevServer": false,
    "isProduction": false,
    "value": "test",
  },
  "files": Object {
    "copy": "+(images|media)/**/*",
  },
  "paths": Object {
    "input": "./assets/",
    "output": "./public/assets/",
    "public": "/assets/",
    "root": "<tmp>",
  },
  "proxy": Object {
    "/": Object {
      "autoRewrite": true,
      "changeOrigin": true,
      "target": "http://localhost:8080",
    },
  },
  "resolve": Object {
    "alias": Object {
      "@": "@{paths.input}",
      "~": "@{paths.input}/scripts",
    },
  },
  "scss": Object {
    "browsers": Array [
      "last 1 version",
    ],
  },
}
`;

exports[`webpack.config.js options to build webpack.config.js matches snapshot for Single Page Application 1`] = `
Object {
  "babel": Object {
    "presets": Array [
      Array [
        "env",
        Object {
          "targets": Object {
            "browsers": "@{autoprefixer.browsers}",
          },
          "useBuiltIns": true,
        },
      ],
      "stage-2",
    ],
  },
  "blocks": Object {
    "list": Array [
      [Function ports],
      [Function filename],
      [Function base],
      [Function script],
      [Function style],
      [Function media],
      [Function plugins],
      [Function devServer],
      [Function backEnd],
    ],
    "timeout": 3000,
  },
  "browserSync": Object {
    "files": Array [
      "./index.ejs",
    ],
    "ghostMode": false,
    "open": true,
    "watchOptions": Object {
      "cwd": "@{paths.input}",
      "ignoreInitial": true,
      "ignored": "*.txt",
    },
  },
  "caching": Object {
    "hash": ".[hash:8]",
  },
  "enabled": Object {
    "sourceMap": "@{env.isDevServer}",
  },
  "entry": Object {
    "main": Array [
      "./scripts/main.js",
    ],
  },
  "env": Object {
    "isDevServer": false,
    "isProduction": false,
    "value": "test",
  },
  "files": Object {
    "copy": "+(images|media)/**/*",
  },
  "html": Object {
    "showErrors": true,
    "template": "./index.ejs",
  },
  "paths": Object {
    "input": "./app/",
    "output": "./public/",
    "public": "/",
    "root": "<tmp>",
  },
  "resolve": Object {
    "alias": Object {
      "@": "@{paths.input}",
      "~": "@{paths.input}/scripts",
    },
  },
  "scss": Object {
    "browsers": Array [
      "last 1 version",
    ],
  },
}
`;
